interfaces/builtin/microstack_support.go

```diff
@@ -0,0 +1,274 @@
+/proc/version r,
+/proc/*/stat r,
+/proc/*/mounts r,
+/proc/*/status r,
+/proc/*/ns/net r,
+
+
+# Used by libvirt to work with network devices created for VMs.
+# E.g. getting operational state and speed of tap devices.
+/sys/class/net/tap*/* rw,
+
+# for usb access
+/dev/bus/usb/ r,
+/sys/bus/ r,
+/sys/class/ r,
+# For hostdev access. The actual devices will be added dynamically
+/sys/bus/usb/devices/ r,
+/sys/devices/**/usb[0-9]*/** r,
+# libusb needs udev data about usb devices (~equal to content of lsusb -v)
+/run/udev/data/+usb* r,
+/run/udev/data/c16[6,7]* r,
+/run/udev/data/c18[0,8,9]* r,
+
+# Libvirt needs access to the PCI config space in order to be able to reset devices.
+/sys/devices/pci*/**/config rw,
+
+# for vfio hotplug on systems without static vfio (LP: #1775777)
+/dev/vfio/vfio rw,
+/dev/vfio/* rw,
+
+# Allow querying the max number of segments for a block device by Libvirt
+# (see QEMU commit 9103f1ce and Linux/block/queue-sysfs.txt).
+/sys/devices/**/block/*/queue/max_segments r,
+
+# required by libpmem init to fts_open()/fts_read() the symlinks in
+# /sys/bus/nd/devices
+/ r, # harmless on any lsb compliant system
+/sys/bus/nd/devices/{,**/} r,
+
+# For ppc device-tree access by Libvirt.
+/proc/device-tree/ r,
+/proc/device-tree/** r,
+/sys/firmware/devicetree/** r,
+
+# "virsh console" support
+/dev/pts/* rw,
+# Used by libvirt.
+/dev/ptmx rw,
+# spice
+owner /{dev,run}/shm/spice.* rw,
+
+# Used by libvirt to create lock files for /dev/pts/<num> devices
+# when handling virsh console access requests.
+/run/lock/ r,
+/run/lock/snap.@{SNAP_INSTANCE_NAME}/ rw,
+/run/lock/snap.@{SNAP_INSTANCE_NAME}/** mrwklix,
+
+# allow connect with openGraphicsFD to work
+unix (send, receive) type=stream addr=none peer=(label=libvirtd),
+unix (send, receive) type=stream addr=none peer=(label=/snap/microstack/current/usr/sbin/libvirtd),
+
+# Allow running utility processes under the specialized AppArmor profiles.
+# These profiles will prevent utility processes escaping confinement.
+capability mac_admin,
+
+# MicroStack services such as libvirt use has a server/client design where
+# unix sockets are used for IPC.
+capability chown,
+
+# MicroStack will also use privilege separation when running utility processes
+capability setuid,
+capability setgid,
+
+# Required by Nova and Neutron (e.g. to create network namespaces), see oslo.privsep docs.
+capability sys_admin,
+# Required by Neutron and OpenvSwitch.
+capability net_admin,
+capability net_broadcast,
+# Required by Nova.
+capability dac_override,
+capability dac_read_search,
+capability fowner,
+
+# Used by libvirt to alter process capabilities via prctl.
+capability setpcap,
+# Used by libvirt to create device special files.
+capability mknod,
+# Used by libvirt to write log messages to the kernel audit log.
+capability audit_write,
+# Used by QEMU for memory locking.
+capability ipc_lock,
+
+# Used my mysql.
+capability sys_nice,
+
+# ptrace
+capability sys_ptrace,
```

Comment:
Depending on what application is using this, you will get this by plugging network-control. If the app in your snap doesn't need it, we'll need to add it somewhere else (perhaps here). What is using it?